src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs

```diff
@@ -0,0 +1,101 @@
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    [ExportInterceptingPropertyValueProvider(PackageIconPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class PackageIconValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private const string PackageIconPropertyName = "PackageIcon";
+        private const string PackPropertyName = "Pack";
+        private const string PackagePathPropertyName = "PackagePath";
+        private const string NoneItem = "None";
+        private const string TrueValue = "True";
+
+        private readonly IProjectItemProvider _sourceItemsProvider;
+        private readonly UnconfiguredProject _unconfiguredProject;
+
+        [ImportingConstructor]
+        public PackageIconValueProvider(
+            [Import(ExportContractNames.ProjectItemProviders.SourceFiles)] IProjectItemProvider sourceItemsProvider,
+            UnconfiguredProject unconfiguredProject)
+        {
+            _sourceItemsProvider = sourceItemsProvider;
+            _unconfiguredProject = unconfiguredProject;
+        }
+
+        // https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packageicon
+        private static string CreatePackageIcon(string filePath, string? packagePath)
+        {
+            string filename = Path.GetFileName(filePath);
+            // Make a slash-only value into empty string, so it won't get prepended onto the path.
+            packagePath = ("\\".Equals(packagePath, StringComparison.Ordinal) || "/".Equals(packagePath, StringComparison.Ordinal))
+                ? string.Empty : (packagePath ?? string.Empty);
+            // The assumption is that PackagePath does not contain a path to a file; only a directory path.
+            return Path.Combine(packagePath, filename);
+        }
+
+        private async Task<IProjectItem?> GetExistingNoneItemAsync(string existingPackageIcon)
+        {
+            foreach (IProjectItem noneItem in await _sourceItemsProvider.GetItemsAsync(NoneItem))
+            {
+                string pack = await noneItem.Metadata.GetEvaluatedPropertyValueAsync(PackPropertyName);
+                string itemIconFilename = Path.GetFileName(noneItem.EvaluatedInclude);
+                string existingIconFilename = Path.GetFileName(existingPackageIcon);
+                // Instead of doing pure equality between itemPackageIcon and existingPackageIcon, a user may update the PackagePath
+                // of the item and forget to update the PackageIcon to reflect those changes, or vice versa. Instead, if the filename
+                // of this packed None item and the filename of the PackageIcon match, consider those to be related to one another.
+                if (TrueValue.Equals(pack, StringComparison.OrdinalIgnoreCase) && itemIconFilename.Equals(existingIconFilename, StringComparison.Ordinal))
```

Comment:
Minor, but consider parsing rather than comparing boolean string values:

```suggestion
                if (bool.TryParse(pack, out bool packBool) && packBool  && itemIconFilename.Equals(existingIconFilename, StringComparison.Ordinal))
```

The parse function will handle any leading/trailing white space correctly. That is, `" true "` is still true.

Reply:
Makes sense. I'm used to there being no direct relationship between the value stored in a data format language and a CLR type. So, I'd normally not use a CLR parser to evaluate a value stored in an XML file. But, I understand how it likely makes sense here since it is MSBuild information. Fixed.